libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Views/SFSDKWindow.m

```diff
@@ -0,0 +1,109 @@
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#import "SFSDKWindow.h"
+
+@implementation SFSDKRootController
+
+-(UIStatusBarStyle)preferredStatusBarStyle
+{
+    UIViewController *topViewController = [SFSDKRootController topViewController:self];
+    UIStatusBarStyle statusBarStyle = UIStatusBarStyleDefault;
+    if (topViewController && topViewController!=self) {
+       statusBarStyle = [topViewController preferredStatusBarStyle];
+    }
+    return statusBarStyle;
+}
+
+-(UIViewController *)childViewControllerForStatusBarStyle
+{
+    UIViewController *topViewController = [SFSDKRootController topViewController:self];
+    if (topViewController && topViewController!=self) {
+        return [topViewController childViewControllerForStatusBarStyle];
+    }
+    return nil;
+}
+
+-(UIViewController *)childViewControllerForStatusBarHidden {
+    
+    UIViewController *topViewController = [SFSDKRootController topViewController:self];
+    if (topViewController && topViewController!=self) {
+        return [topViewController childViewControllerForStatusBarHidden];
+    }
+    return nil;
+}
+
+-(BOOL)shouldAutorotate
+{
+     UIViewController *topViewController = [SFSDKRootController topViewController:self];
+    if (topViewController!=nil && topViewController!=self)
+        return [topViewController shouldAutorotate];
+    return YES;
+}
+
+- (UIInterfaceOrientationMask)supportedInterfaceOrientations
+{
+    UIViewController *topViewController = [SFSDKRootController topViewController:self];
+    if (topViewController!=nil && topViewController!=self)
+         return [topViewController supportedInterfaceOrientations];
+    
+    return UIInterfaceOrientationMaskAll;
+}
+
+#pragma mark - Helper class methods
++ (UIViewController *)topViewController:(SFSDKRootController *) controller
+{
+    UIViewController *topViewController = controller;
+    while (topViewController.presentedViewController != nil
+           && !topViewController.presentedViewController.isBeingDismissed) {
+        //stop if we find that an alert has been presented
+        if ([topViewController.presentedViewController isKindOfClass:[UIAlertController class]]) {
+            break;
+        }
+        topViewController = topViewController.presentedViewController;
+    }
+    return topViewController;
+}
+@end
+
+@interface SFSDKWindow()
+
+@end
+
+@implementation SFSDKWindow
+
+- (instancetype) initWithFrame:(CGRect) frame {
+    self = [super initWithFrame:frame];
+    
+    if (self) {
+        self.hidden = YES;
```

Comment:
It seems like overkill to manage a child class of `UIWindow`, simply for setting its default visibility.  Can we not just use `UIWindow` and configure it in this case?